src/shared/dissect-image.c

```diff
@@ -2648,7 +2648,9 @@ static int do_crypt_activate_verity(
                 /* So this didn't work via the kernel, then let's try userspace validation instead. If that
                  * works we'll try to activate without telling the kernel the signature. */
 
-                r = validate_signature_userspace(verity);
+                /* Propagate the original kernel error, so that the fallback logic can work, as the
+                 * device-mapper is finicky around concurrent activations of the same volume */
+                RET_GATHER(r, validate_signature_userspace(verity));
```

Comment:
this doesn't look right in case HAVE_CRYPT_ACTIVATE_BY_SIGNED_KEY is unset.

because `r` isn't initialized to anything we want to gather here in that case. hence in the `#else` branch above set r = 0 explicitly

Reply:
or hnmm, actually maybe `r = -EOPNOTSUPP` rather than `r = 0` in the `#else` branch?